File: 202312/1578.go
func minCost(colors string, neededTime []int) int {
	ans := 0
	p := '#'
	curr := 0
	total := 0
	for k, c := range colors {
		total += neededTime[k]
		if p != c {
			ans += curr
			curr = neededTime[k]
		} else {
			curr = max(curr, neededTime[k])
		}
		p = c
	}
	ans += curr
	return total - ans
}

File: 202511/1578.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        ans = 0
        cc = " "
        total, mi = 0, 0
        for i, color in enumerate(colors):
            if color != cc:
                ans += total - mi
                cc = color
                total = neededTime[i]
                mi = total
            else:
                total += neededTime[i]
                mi = max(mi, neededTime[i])
        return ans + total - mi
